src/components/course/data/hooks.js

```diff
@@ -168,33 +167,59 @@ export function useCoursePriceForUserSubsidy(activeCourseRun, userSubsidy) {
         return null;
       }
 
-      const priceDetails = {
-        list: numberWithPrecision(listPrice),
+      const onlyListPrice = {
+        list: listPrice,
       };
-      if (!userSubsidy) {
-        return priceDetails;
-      }
 
-      const { discountType, discountValue } = userSubsidy;
-      let discountedPrice;
+      // Case 1: License Subsidy
+      if (userSubsidy && userSubsidy.discountValue) {
+        const { discountType, discountValue } = userSubsidy;
+        let discountedPrice;
+
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
+          discountedPrice = listPrice - (listPrice * (discountValue / 100));
+        }
+
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
+          discountedPrice = Math.max(listPrice - discountValue, 0);
+        }
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
-        discountedPrice = listPrice - (listPrice * (discountValue / 100));
+        if (isDefinedAndNotNull(discountedPrice)) {
+          return {
+            ...onlyListPrice,
+            discounted: discountedPrice,
+          };
+        }
+        return {
+          ...onlyListPrice,
+          discounted: onlyListPrice.list,
+        };
       }
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
-        discountedPrice = Math.max(listPrice - discountValue, 0);
+      // Case 2: No subsidy, no offers for course
+      if (!offers) {
+        return onlyListPrice;
       }
 
-      if (isDefinedAndNotNull(discountedPrice)) {
-        priceDetails.discounted = numberWithPrecision(discountedPrice);
-      } else {
-        priceDetails.discounted = priceDetails.list;
+      const offerForCourse = findOfferForCourse(offers, catalogList);
+
+      if (!offerForCourse) {
+        return onlyListPrice;
       }
 
-      return priceDetails;
+      // Case 3: No subsidy, but found offer for course
+      const { benefitValue, usageType } = offerForCourse;
+      if (usageType === 'Percentage') {
+        // only 100% discount supported as of now
+        const discountedPrice = listPrice - (listPrice * (benefitValue / 100));
+        return {
+          ...onlyListPrice,
+          discounted: discountedPrice,
+        };
+      }
+      return onlyListPrice;
     },
-    [activeCourseRun, userSubsidy],
+    [activeCourseRun, userSubsidy, offers],
```

Comment:
`catalogList` should probably also be included

Reply:
yes, thanks!